packages/utils/src/structs/checkpoint.cairo

```diff
@@ -3,14 +3,17 @@
 
 use core::num::traits::Sqrt;
 use crate::math;
+use starknet::storage::{StoragePath, StorageAsPath, Vec, VecTrait, Mutable, MutableVecTrait};
+use starknet::storage::{StoragePointerReadAccess, StoragePointerWriteAccess};
 use starknet::storage_access::StorePacking;
-use super::storage_array::{StorageArray, StorageArrayTrait};
+
+const _2_POW_184: felt252 = 0x10000000000000000000000000000000000000000000000;
 
 /// `Trace` struct, for checkpointing values as they change at different points in
 /// time, and later looking up past values by block timestamp.
-#[derive(Copy, Drop, starknet::Store)]
+#[starknet::storage_node]
 pub struct Trace {
-    pub checkpoints: StorageArray<Checkpoint>
+    pub checkpoints: Vec<Checkpoint>
```

Comment:
We need to add this breaking change to the CHANGELOG. A sub-item of the ERC20Votes refactor seems fit.

Reply:
Also, and entry for the removal of `StorageArray` from the library.

Reply:
Did both! Let me know what you think